packages/react-router/modules/matchPath.js

```diff
@@ -26,11 +26,17 @@ const compilePath = (pattern, options) => {
 /**
  * Public API for matching a URL pathname to a path pattern.
  */
-const matchPath = (pathname, path, options = {}) => {
+const matchPath = (pathname, path, options = {}, parentMatch = null) => {
   const { exact = false, strict = false } = options
 
-  if (!path)
-    return { url: pathname, isExact: true, params: {} }
+  const absolute = isAbsolute(path)
+
+  if (path === undefined)
+    return { url: parentMatch ? parentMatch.url : '/', isExact: true, params: {} }
```

Comment:
I guess that I'm not really sure why someone who is using a pathless `<Route>` would be looking at `match.url`. If they need the full path, `location.pathname` already provides that. Maybe it would make sense for the returned `match` object to not have a `url` property.